device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/https/HttpsTransport.java

```diff
@@ -324,6 +324,17 @@ public boolean isEmpty()
     }
 
     /**
+     * Registers a callback to be executed whenever the https connection is lost or established.
+     * 
+     * @param callback the callback to be called.
+     * @param callbackContext a context to be passed to the callback. Can be
+     * {@code null} if no callback is provided.
+     */
+    public void registerConnectionStateCallback(IotHubConnectionStateCallback callback, Object callbackContext) {
+        //HTTPS is stateless so this should be a no-op
```

Comment:
Yes.  We probably should not allow someone to register a callback on this type of connection.